// Copyright 2016 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// -----------------------------------------------------------------------------

syntax = "proto3";
option java_multiple_files = true;
option java_package = "sawtooth.sdk.protobuf";


// The registration request from a peer to the validator
message PeerRegisterRequest {
    string identity = 1;
}

// The unregistration request from a peer to the validator
message PeerUnregisterRequest {
    string identity = 1;
}

message PingRequest {
}

message GossipMessage {
    bytes content = 1;
    string content_type = 2;
}

// A response sent from the validator to the peer acknowledging message
// receipt
message NetworkAcknowledgement {
    enum Status {
        OK = 0;
        ERROR = 1;
    }

    Status status = 1;
}
